<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Processing Hi-C paired-end fastq files in R — HiCool • HiCool</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Processing Hi-C paired-end fastq files in R — HiCool"><meta property="og:description" content="HiCool::HiCool() automatically processes paired-end HiC sequencing files
by performing the following steps:
Automatically setting up an appropriate conda environment using basilisk;
Mapping the reads to the provided genome reference using hicstuff and filtering of irrelevant pairs;
Filtering the resulting pairs file to remove unwanted chromosomes (e.g. chrM);
Binning the filtered pairs into a cool file at a chosen resolution;
Generating a multi-resolution mcool file;
Normalizing matrices at each resolution by iterative corretion using cooler.


The filtering strategy used by hicstuff is described in Cournac et al., BMC Genomics 2012."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">HiCool</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/HiCool.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/js2264/HiCool/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Processing Hi-C paired-end fastq files in R</h1>
    <small class="dont-index">Source: <a href="https://github.com/js2264/HiCool/blob/HEAD/R/HiCool.R" class="external-link"><code>R/HiCool.R</code></a>, <a href="https://github.com/js2264/HiCool/blob/HEAD/R/utils.R" class="external-link"><code>R/utils.R</code></a></small>
    <div class="hidden name"><code>HiCool.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>HiCool::HiCool()</code> automatically processes paired-end HiC sequencing files
by performing the following steps:</p><ol><li><p>Automatically setting up an appropriate conda environment using basilisk;</p></li>
<li><p>Mapping the reads to the provided genome reference using <code>hicstuff</code> and filtering of irrelevant pairs;</p></li>
<li><p>Filtering the resulting pairs file to remove unwanted chromosomes (e.g. chrM);</p></li>
<li><p>Binning the filtered pairs into a cool file at a chosen resolution;</p></li>
<li><p>Generating a multi-resolution mcool file;</p></li>
<li><p>Normalizing matrices at each resolution by iterative corretion using cooler.</p></li>
</ol><p>The filtering strategy used by <code>hicstuff</code> is described in Cournac et al., BMC Genomics 2012.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">HiCool</span><span class="op">(</span></span>
<span>  <span class="va">r1</span>,</span>
<span>  <span class="va">r2</span>,</span>
<span>  <span class="va">genome</span>,</span>
<span>  restriction <span class="op">=</span> <span class="st">"DpnII,HinfI"</span>,</span>
<span>  resolutions <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  iterative <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  balancing_args <span class="op">=</span> <span class="st">" --min-nnz 10 --mad-max 5 "</span>,</span>
<span>  threads <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  exclude_chr <span class="op">=</span> <span class="st">"Mito|chrM|MT"</span>,</span>
<span>  output <span class="op">=</span> <span class="st">"HiCool"</span>,</span>
<span>  keep_bam <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  build_report <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  scratch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">importHiCoolFolder</span><span class="op">(</span><span class="va">output</span>, <span class="va">hash</span>, resolution <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">getHiCoolArgs</span><span class="op">(</span><span class="va">log</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">getHicStats</span><span class="op">(</span><span class="va">log</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>r1</dt>
<dd><p>Path to fastq file (R1 read)</p></dd>


<dt>r2</dt>
<dd><p>Path to fastq file (R2 read)</p></dd>


<dt>genome</dt>
<dd><p>Genome used to map the reads on, provided either
as a fasta file (in which case the bowtie2 index will be automatically
generated), or as a prefix to a bowtie2 index (e.g. <code>mm10</code> for
<code>mm10.*.bt2</code> files). Genome can also be a unique ID for the following
references: <code>hg38</code>, <code>mm10</code>, <code>dm6</code>, <code>R64-1-1</code>, <code>GRZc10</code>, <code>WBcel235</code>,
<code>Galgal4</code>.</p></dd>


<dt>restriction</dt>
<dd><p>Restriction enzyme(s) used in HiC (Default: "DpnII,HinfI")</p></dd>


<dt>resolutions</dt>
<dd><p>Resolutions used to bin the final mcool file
(Default: 5 levels of resolution automatically inferred according to genome size)</p></dd>


<dt>iterative</dt>
<dd><p>Should the read mapping be performed iteratively?
(Default: TRUE)</p></dd>


<dt>balancing_args</dt>
<dd><p>Balancing arguments for cooler.
See <code>cooler</code> documentation <a href="https://cooler.readthedocs.io/en/latest/cli.html#cooler-balance" class="external-link">here</a>
for a list of all available balancing arguments.
These defaults match those used by the 4DN consortium.</p></dd>


<dt>threads</dt>
<dd><p>Number of CPUs used for parallelization. (Default: 1)</p></dd>


<dt>exclude_chr</dt>
<dd><p>Chromosomes excluded from the final .mcool file. This will
not affect the pairs file. (Default: "Mito|chrM|MT")</p></dd>


<dt>output</dt>
<dd><p>Output folder used by HiCool.</p></dd>


<dt>keep_bam</dt>
<dd><p>Should the bam files be kept? (Default: FALSE)</p></dd>


<dt>build_report</dt>
<dd><p>Should an automated report be computed? (Default: TRUE)</p></dd>


<dt>scratch</dt>
<dd><p>Path to temporary directory where processing will take place.
(Default: <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code>)</p></dd>


<dt>hash</dt>
<dd><p>Unique 6-letter ID used to identify files from a specific
HiCool processing run.</p></dd>


<dt>resolution</dt>
<dd><p>Resolution used to import the mcool file</p></dd>


<dt>log</dt>
<dd><p>Path to log file generated by hicstuff/hicool</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A <code>CoolFile</code> object with prefilled <code>pairsFile</code> and <code>metadata</code> slots.</p>
    </div>
    <div id="hicool-utils">
    <h2>HiCool utils</h2>
    

<ul><li><p><code>importHiCoolFolder(folder, hash)</code> automatically finds the different processed files
associated with a specific HiCool::HiCool() processing hash ID.</p></li>
<li><p>getHiCoolArgs() parses the log file generated by HiCool::HiCool() during
processing to recover which arguments were used.</p></li>
<li><p>getHicStats() parses the log file generated by HiCool::HiCool() during
processing to recover pre-computed stats about pair numbers, filtering
thresholds, etc.</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">r1</span> <span class="op">&lt;-</span> <span class="fu">HiContactsData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiContactsData/man/HiContactsData.html" class="external-link">HiContactsData</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="st">'yeast_wt'</span>, format <span class="op">=</span> <span class="st">'fastq_R1'</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> snapshotDate(): 2023-02-28</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?HiContactsData and browseVignettes('HiContactsData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span><span class="va">r2</span> <span class="op">&lt;-</span> <span class="fu">HiContactsData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiContactsData/man/HiContactsData.html" class="external-link">HiContactsData</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="st">'yeast_wt'</span>, format <span class="op">=</span> <span class="st">'fastq_R2'</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> snapshotDate(): 2023-02-28</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?HiContactsData and browseVignettes('HiContactsData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span><span class="va">hcf</span> <span class="op">&lt;-</span> <span class="fu">HiCool</span><span class="op">(</span><span class="va">r1</span>, <span class="va">r2</span>, genome <span class="op">=</span> <span class="st">'R64-1-1'</span>, output <span class="op">=</span> <span class="st">'./HiCool/'</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> HiCool :: Recovering bowtie2 genome index from AWS iGenomes...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> HiCool :: Initiating processing of fastq files [tmp folder: /tmp/RtmpPdwOTI/4O1XMN]...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> HiCool :: Mapping fastq files...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> HiCool :: Best-suited minimum resolution automatically inferred: 1000</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> HiCool :: Removing unwanted chromosomes...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> HiCool :: Parsing pairs into .cool file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> HiCool :: Generating multi-resolution .mcool file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> HiCool :: Balancing .mcool file...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> HiCool :: Tidying up everything for you...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> HiCool :: .fastq to .mcool processing done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> HiCool :: Check ./HiCool/folder to find the generated files</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> HiCool :: Generating HiCool report. This might take a while.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> HiCool :: Report generated and available @ /__w/HiCool/HiCool/docs/reference/HiCool/391a6cedce1_7833^mapped-R64-1-1^4O1XMN.html</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> HiCool :: All processing successfully achieved. Congrats!</span>
<span class="r-in"><span><span class="va">hcf</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CoolFile object</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> .mcool file: ./HiCool//matrices/391a6cedce1_7833^mapped-R64-1-1^4O1XMN.mcool </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> resolution: 1000 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pairs file: ./HiCool//pairs/391a6cedce1_7833^mapped-R64-1-1^4O1XMN.pairs </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(3): log args stats</span>
<span class="r-in"><span><span class="fu">getHiCoolArgs</span><span class="op">(</span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">hcf</span><span class="op">)</span><span class="op">$</span><span class="va">log</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $r1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/github/home/.cache/R/ExperimentHub/391a6cedce1_7833"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $r2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/github/home/.cache/R/ExperimentHub/391a264bc8eb_7834"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $genome</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/tmp/RtmpPdwOTI/R64-1-1"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $resolutions</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $restriction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "DpnII,HinfI"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $iterative</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $balancing_args</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] " --min-nnz 10 --mad-max 5 "</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $threads</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "./HiCool/"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $exclude_chr</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Mito|chrM|MT"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $keep_bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $scratch</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/tmp/RtmpPdwOTI"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $wd</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/__w/HiCool/HiCool/docs/reference"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="fu">getHicStats</span><span class="op">(</span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">hcf</span><span class="op">)</span><span class="op">$</span><span class="va">log</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $nFragments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 1e+05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $nPairs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 73993</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $nDangling</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 10027</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $nSelf</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 2205</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $nDumped</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 83</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $nFiltered</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 61678</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $nDups</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 719</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $nUnique</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 60959</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $threshold_uncut</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $threshold_self</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/readLines.html" class="external-link">readLines</a></span><span class="op">(</span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">hcf</span><span class="op">)</span><span class="op">$</span><span class="va">log</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "HiCool working directory ::: /__w/HiCool/HiCool/docs/reference"                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [2] "HiCool argument ::: r1: /github/home/.cache/R/ExperimentHub/391a6cedce1_7833"                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [3] "HiCool argument ::: r2: /github/home/.cache/R/ExperimentHub/391a264bc8eb_7834"                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] "HiCool argument ::: genome: /tmp/RtmpPdwOTI/R64-1-1"                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [5] "HiCool argument ::: resolutions: "                                                               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [6] "HiCool argument ::: restriction: DpnII,HinfI"                                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [7] "HiCool argument ::: iterative: TRUE"                                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [8] "HiCool argument ::: balancing_args:  --min-nnz 10 --mad-max 5 "                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [9] "HiCool argument ::: threads: 1"                                                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [10] "HiCool argument ::: output: ./HiCool/"                                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [11] "HiCool argument ::: exclude_chr: Mito|chrM|MT"                                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [12] "HiCool argument ::: keep_bam: FALSE"                                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [13] "HiCool argument ::: scratch: /tmp/RtmpPdwOTI"                                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [14] "----------------"                                                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [15] "## hicstuff: v3.1.5 log file"                                                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16] "## date: 2023-03-12 18:55:55"                                                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [17] "## enzyme: DpnII,HinfI"                                                                          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [18] "## input1: /github/home/.cache/R/ExperimentHub/391a6cedce1_7833 "                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [19] "## input2: /github/home/.cache/R/ExperimentHub/391a264bc8eb_7834"                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [20] "## ref: /tmp/RtmpPdwOTI/R64-1-1"                                                                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [21] "---"                                                                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [22] "2023-03-12,18:55:58 :: INFO :: 100000 reads to parse"                                            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [23] "2023-03-12,18:55:58 :: INFO :: Truncating unaligned reads to 20bp and mapping."                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [24] "2023-03-12,18:56:01 :: INFO :: 19256 reads left to map."                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [25] "2023-03-12,18:56:01 :: INFO :: Trying to map unaligned reads at full length (35bp)."             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [26] "2023-03-12,18:56:02 :: INFO :: 18387 reads left to map."                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [27] "2023-03-12,18:56:02 :: INFO :: 81613 reads aligned / 100000 total reads."                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [28] "2023-03-12,18:56:03 :: INFO :: 100000 reads to parse"                                            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [29] "2023-03-12,18:56:03 :: INFO :: Truncating unaligned reads to 20bp and mapping."                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [30] "2023-03-12,18:56:06 :: INFO :: 20617 reads left to map."                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [31] "2023-03-12,18:56:06 :: INFO :: Trying to map unaligned reads at full length (35bp)."             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [32] "2023-03-12,18:56:06 :: INFO :: 19373 reads left to map."                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [33] "2023-03-12,18:56:07 :: INFO :: 80627 reads aligned / 100000 total reads."                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [34] "2023-03-12,18:56:11 :: INFO :: 81% reads (single ends) mapped with Q &gt;= 10.0 (162240/200000)"    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [35] "2023-03-12,18:56:15 :: INFO :: Filtering with thresholds: uncuts=7 loops=7"                      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [36] "2023-03-12,18:56:15 :: INFO :: Proportion of inter contacts: 22.03% (intra: 48089, inter: 13589)"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [37] "2023-03-12,18:56:15 :: INFO :: 12315 pairs discarded: Loops: 2205, Uncuts: 10027, Weirds: 83"    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [38] "2023-03-12,18:56:15 :: INFO :: 61678 pairs kept (83.36%)"                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [39] "2023-03-12,18:56:18 :: INFO :: 1% PCR duplicates have been filtered out (719 / 61678 pairs) "    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [40] "2023-03-12,18:56:20 :: INFO :: Contact map generated after 0h 0m 25s"                            </span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Jacques Serizay.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

