<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>HiCool • HiCool</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="HiCool">
<meta property="og:description" content="HiCool">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">HiCool</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/HiCool.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/js2264/HiCool/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>HiCool</h1>
                        <h4 data-toc-skip class="author">Jacques
Serizay</h4>
            
            <h4 data-toc-skip class="date">2023-04-04</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/js2264/HiCool/blob/HEAD/vignettes/HiCool.Rmd" class="external-link"><code>vignettes/HiCool.Rmd</code></a></small>
      <div class="hidden name"><code>HiCool.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="processing-sequencing-hi-c-libraries-with-hicool">Processing sequencing Hi-C libraries with <code>HiCool</code><a class="anchor" aria-label="anchor" href="#processing-sequencing-hi-c-libraries-with-hicool"></a>
</h2>
<p>The <code>HiCool</code> R/Bioconductor package provides an
<strong>end-to-end interface</strong> to process and normalize Hi-C
paired-end fastq reads into <code>.(m)cool</code> files.</p>
<ol style="list-style-type: decimal">
<li>The heavy lifting (fastq mapping, pairs parsing and pairs filtering)
is performed by the underlying lightweight <code>hicstuff</code> python
library (<a href="https://github.com/koszullab/hicstuff" class="external-link">https://github.com/koszullab/hicstuff</a>).</li>
<li>Pairs filering is done using the approach described in <a href="https://doi.org/10.1186/1471-2164-13-436" class="external-link">Cournac et al., 2012</a>
and implemented in <code>hicstuff</code>.</li>
<li>
<code>cooler</code> (<a href="https://github.com/open2c/cooler" class="external-link">https://github.com/open2c/cooler</a>)
library is used to parse pairs into a multi-resolution, balanced
<code>.mcool</code> file. <code>.(m)cool</code> is a compact, indexed
HDF5 file format specifically tailored for efficiently storing HiC-based
data. The <code>.(m)cool</code> file format was developed by Abdennur
and Mirny and <a href="https://doi.org/10.1093/bioinformatics/btz540" class="external-link">published in
2019</a>.</li>
<li>Internally, all these external dependencies are automatically
installed and managed in R by a <code>basilisk</code> environment.</li>
</ol>
<p>The main processing function offered in this package is
<code><a href="../reference/HiCool.html">HiCool()</a></code>. To process <code>.fastq</code> reads into
<code>.pairs</code> &amp; <code>.mcool</code> files, one needs to
provide:</p>
<ul>
<li>The path to each fastq file (<code>r1</code> and
<code>r2</code>);</li>
<li>The genome reference, as a path to a <code>.fasta</code> sequence
file, a path to a pre-computed <code>bowtie2</code> index or a supported
ID character (<code>hg38</code>, <code>mm10</code>, <code>dm6</code>,
<code>R64-1-1</code>, <code>WBcel235</code>, <code>GRCz10</code>,
<code>Galgal4</code>);</li>
<li>The restriction enzyme(s) used for Hi-C.</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/HiCool.html">HiCool</a></span><span class="op">(</span></span>
<span>    r1 <span class="op">=</span> <span class="st">'&lt;PATH-TO-R1.fq.gz&gt;'</span>, </span>
<span>    r2 <span class="op">=</span> <span class="st">'&lt;PATH-TO-R2.fq.gz&gt;'</span>, </span>
<span>    restriction <span class="op">=</span> <span class="st">'&lt;RE1(,RE2)&gt;'</span>, </span>
<span>    resolutions <span class="op">=</span> <span class="st">"&lt;resolutions of interest&gt;"</span>, </span>
<span>    genome <span class="op">=</span> <span class="st">'&lt;GENOME_ID&gt;'</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Here is a concrete example of Hi-C data processing.</p>
<ul>
<li>Example fastq files are retrieved using the
<code>HiContactsData</code> package.</li>
<li>Two restriction enzymes are used (these are the enzymes used in the
Arima Kit).</li>
<li>The final <code>.mcool</code> file will have three levels of
resolutions, from 1000bp to 8000bp.</li>
<li>The data will be mapped on <code>R64-1-1</code>, the yeast genome
reference.</li>
<li>All output files will be placed in <code>output/</code>
directory.</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/js2264/HiCool" class="external-link">HiCool</a></span><span class="op">)</span></span>
<span><span class="va">hcf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/HiCool.html">HiCool</a></span><span class="op">(</span></span>
<span>    r1 <span class="op">=</span> <span class="fu">HiContactsData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiContactsData/man/HiContactsData.html" class="external-link">HiContactsData</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="st">'yeast_wt'</span>, format <span class="op">=</span> <span class="st">'fastq_R1'</span><span class="op">)</span>, </span>
<span>    r2 <span class="op">=</span> <span class="fu">HiContactsData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiContactsData/man/HiContactsData.html" class="external-link">HiContactsData</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="st">'yeast_wt'</span>, format <span class="op">=</span> <span class="st">'fastq_R2'</span><span class="op">)</span>, </span>
<span>    restriction <span class="op">=</span> <span class="st">'DpnII,HinfI'</span>, </span>
<span>    resolutions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4000</span>, <span class="fl">8000</span>, <span class="fl">16000</span><span class="op">)</span>, </span>
<span>    genome <span class="op">=</span> <span class="st">'R64-1-1'</span>, </span>
<span>    output <span class="op">=</span> <span class="st">'./HiCool/'</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; snapshotDate(): 2023-04-03</span></span>
<span><span class="co">#&gt; see ?HiContactsData and browseVignettes('HiContactsData') for documentation</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="co">#&gt; snapshotDate(): 2023-04-03</span></span>
<span><span class="co">#&gt; see ?HiContactsData and browseVignettes('HiContactsData') for documentation</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="co">#&gt; HiCool :: Recovering bowtie2 genome index from AWS iGenomes...</span></span>
<span><span class="co">#&gt; HiCool :: Initiating processing of fastq files [tmp folder: /tmp/Rtmpk8UZfO/C71OG5]...</span></span>
<span><span class="co">#&gt; HiCool :: Mapping fastq files...</span></span>
<span><span class="co">#&gt; HiCool :: Removing unwanted chromosomes...</span></span>
<span><span class="co">#&gt; HiCool :: Parsing pairs into .cool file...</span></span>
<span><span class="co">#&gt; HiCool :: Generating multi-resolution .mcool file...</span></span>
<span><span class="co">#&gt; HiCool :: Balancing .mcool file...</span></span>
<span><span class="co">#&gt; HiCool :: Tidying up everything for you...</span></span>
<span><span class="co">#&gt; HiCool :: .fastq to .mcool processing done!</span></span>
<span><span class="co">#&gt; HiCool :: Check ./HiCool/folder to find the generated files</span></span>
<span><span class="co">#&gt; HiCool :: Generating HiCool report. This might take a while.</span></span>
<span><span class="co">#&gt; HiCool :: Report generated and available @ /__w/HiCool/HiCool/vignettes/HiCool/39bf267dd9d3_7833^mapped-R64-1-1^C71OG5.html</span></span>
<span><span class="co">#&gt; HiCool :: All processing successfully achieved. Congrats!</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hcf</span></span>
<span><span class="co">#&gt; CoolFile object</span></span>
<span><span class="co">#&gt; .mcool file: ./HiCool//matrices/39bf267dd9d3_7833^mapped-R64-1-1^C71OG5.mcool </span></span>
<span><span class="co">#&gt; resolution: 4000 </span></span>
<span><span class="co">#&gt; pairs file: ./HiCool//pairs/39bf267dd9d3_7833^mapped-R64-1-1^C71OG5.pairs </span></span>
<span><span class="co">#&gt; metadata(3): log args stats</span></span>
<span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">hcf</span><span class="op">)</span></span>
<span><span class="co">#&gt; $log</span></span>
<span><span class="co">#&gt; [1] "./HiCool//logs/39bf267dd9d3_7833^mapped-R64-1-1^C71OG5.log"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $args</span></span>
<span><span class="co">#&gt; $args$r1</span></span>
<span><span class="co">#&gt; [1] "/github/home/.cache/R/ExperimentHub/39bf267dd9d3_7833"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $args$r2</span></span>
<span><span class="co">#&gt; [1] "/github/home/.cache/R/ExperimentHub/39bf5b5e4e4_7834"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $args$genome</span></span>
<span><span class="co">#&gt; [1] "/tmp/Rtmpk8UZfO/R64-1-1"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $args$resolutions</span></span>
<span><span class="co">#&gt; [1] "4000"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $args$resolutions</span></span>
<span><span class="co">#&gt; [1] "8000"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $args$resolutions</span></span>
<span><span class="co">#&gt; [1] "16000"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $args$restriction</span></span>
<span><span class="co">#&gt; [1] "DpnII,HinfI"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $args$iterative</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $args$balancing_args</span></span>
<span><span class="co">#&gt; [1] " --min-nnz 10 --mad-max 5 "</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $args$threads</span></span>
<span><span class="co">#&gt; [1] 1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $args$output</span></span>
<span><span class="co">#&gt; [1] "./HiCool/"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $args$exclude_chr</span></span>
<span><span class="co">#&gt; [1] "Mito|chrM|MT"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $args$keep_bam</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $args$scratch</span></span>
<span><span class="co">#&gt; [1] "/tmp/Rtmpk8UZfO"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $args$wd</span></span>
<span><span class="co">#&gt; [1] "/__w/HiCool/HiCool/vignettes"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $stats</span></span>
<span><span class="co">#&gt; $stats$nFragments</span></span>
<span><span class="co">#&gt; [1] 1e+05</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $stats$nPairs</span></span>
<span><span class="co">#&gt; [1] 73993</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $stats$nDangling</span></span>
<span><span class="co">#&gt; [1] 10027</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $stats$nSelf</span></span>
<span><span class="co">#&gt; [1] 2205</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $stats$nDumped</span></span>
<span><span class="co">#&gt; [1] 83</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $stats$nFiltered</span></span>
<span><span class="co">#&gt; [1] 61678</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $stats$nDups</span></span>
<span><span class="co">#&gt; [1] 719</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $stats$nUnique</span></span>
<span><span class="co">#&gt; [1] 60959</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $stats$threshold_uncut</span></span>
<span><span class="co">#&gt; [1] 7</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $stats$threshold_self</span></span>
<span><span class="co">#&gt; [1] 7</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="optional-parameters">Optional parameters<a class="anchor" aria-label="anchor" href="#optional-parameters"></a>
</h2>
<p>Extra optional arguments can be passed to the <code>hicstuff</code>
workhorse library:</p>
<ul>
<li>
<code>iterative</code> <logical> (default: <code>TRUE</code>): By
default, <code>hicstuff</code> first truncates your set of reads to 20bp
and attempts to align the truncated reads, then moves on to aligning
40bp-truncated reads for those which could not be mapped, etc. This
procedure is longer than a traditional mapping but allows for more pairs
to be rescued. Set to <code>FALSE</code> if you want to perform standard
alignment of fastq files without iterative alignment;</logical>
</li>
<li>
<code>balancing_args</code> <character> (default:
<code>" --min-nnz 10 --mad-max 5 "</code>): Specify here any balancing
argument to be used by <code>cooler</code> when normalizing the binned
contact matrices. Full list of options available at <a href="https://cooler.readthedocs.io/en/latest/cli.html#cooler-balance" class="external-link">cooler
documentation website</a>;</character>
</li>
<li>
<code>threads</code> <integer> (default: <code>1L</code>): Number of
CPUs to use to process data;</integer>
</li>
<li>
<code>exclude_chr</code> <character> (default:
<code>'Mito|chrM|MT'</code>): List here any chromosome you wish to
remove from the final contact matrix file;</character>
</li>
<li>
<code>keep_bam</code> <logical> (default: <code>FALSE</code>): Set
to <code>TRUE</code> if you wish to keep the pair of <code>.bam</code>
files;</logical>
</li>
<li>
<code>scratch</code> <character> (default: <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code>):
Points to a temporary directory to be used for processing.</character>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="output-files">Output files<a class="anchor" aria-label="anchor" href="#output-files"></a>
</h2>
<p>The important files generated by <code>HiCool</code> are the
following:</p>
<ul>
<li>A log file:
<code>&lt;output_folder&gt;/logs/&lt;prefix&gt;^mapped-&lt;genome&gt;^&lt;hash&gt;.log</code>
</li>
<li>A multi-resolution, balanced contact matrix file:
<code>&lt;output_folder&gt;/matrices/&lt;prefix&gt;^mapped-&lt;genome&gt;^&lt;hash&gt;.mcool</code>
</li>
<li>A <code>.pairs</code> file:
<code>&lt;output_folder&gt;/pairs/&lt;prefix&gt;^mapped-&lt;genome&gt;^&lt;hash&gt;.pairs</code>
</li>
<li>Several diagnosis plots:
<code>&lt;output_folder&gt;/plots/&lt;prefix&gt;^mapped-&lt;genome&gt;^&lt;hash&gt;_*.pdf</code>.</li>
</ul>
<p>The diagnosis plots illustrate how pairs were filtered during the
processing, using a strategy described in
<code>Cournac et al., BMC Genomics 2012</code>. The
<code>event_distance</code> chart represents the frequency of
<code>++</code>, <code>+-</code>, <code>-+</code> and <code>--</code>
pairs in the library, as a function of the number of restriction sites
between each end of the pairs, and shows the inferred filtering
threshold. The <code>event_distribution</code> chart indicates the
proportion of each type of pairs (e.g. <code>dangling</code>,
<code>uncut</code>, <code>abnormal</code>, …) and the total number of
pairs retained (<code>3D intra</code> + <code>3D inter</code>).</p>
<p><strong>Notes:</strong></p>
<ul>
<li>
<code>.pairs</code> file format is defined by the <a href="https://github.com/4dn-dcic/pairix/blob/master/pairs_format_specification.md" class="external-link">4DN
consortium</a>;</li>
<li>
<code>.(m)cool</code> file format is defined by <code>cooler</code>
authors in the <a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtz540" class="external-link">supporting
publication</a>.</li>
</ul>
</div>
<div class="section level2">
<h2 id="system-dependencies">System dependencies<a class="anchor" aria-label="anchor" href="#system-dependencies"></a>
</h2>
<p>Processing Hi-C sequencing libraries into <code>.pairs</code> and
<code>.mcool</code> files requires several dependencies, to (1) align
reads to a reference genome, (2) manage alignment files (SAM), (3)
filter pairs, (4) bin them to a specific resolution and (5)</p>
<p>All system dependencies are internally managed by
<code>basilisk</code>. <code>HiCool</code> maintains a
<code>basilisk</code> environment containing:</p>
<ul>
<li><code>python 3.9.1</code></li>
<li><code>bowtie2 2.4.5</code></li>
<li><code>samtools 1.7</code></li>
<li><code>hicstuff 3.1.5</code></li>
<li><code>cooler 0.8.11</code></li>
<li><code>chromosight 1.6.3</code></li>
</ul>
<p>The first time <code><a href="../reference/HiCool.html">HiCool()</a></code> is executed, a fresh
<code>basilisk</code> environment will be created and required
dependencies automatically installed. This ensures compatibility between
the different system dependencies needed to process Hi-C fastq
files.</p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R Under development (unstable) (2023-03-16 r83996)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats    graphics utils    methods  base    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] HiContactsData_1.1.9  ExperimentHub_2.7.1   AnnotationHub_3.7.3  </span></span>
<span><span class="co">#&gt; [4] BiocFileCache_2.7.2   dbplyr_2.3.2          BiocGenerics_0.45.3  </span></span>
<span><span class="co">#&gt; [7] HiCool_0.99.5         HiCExperiment_0.99.12 BiocStyle_2.27.1     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] DBI_1.1.3                     bitops_1.0-7                 </span></span>
<span><span class="co">#&gt;   [3] rlang_1.1.0                   magrittr_2.0.3               </span></span>
<span><span class="co">#&gt;   [5] matrixStats_0.63.0            compiler_4.3.0               </span></span>
<span><span class="co">#&gt;   [7] RSQLite_2.3.1                 dir.expiry_1.7.0             </span></span>
<span><span class="co">#&gt;   [9] png_0.1-8                     systemfonts_1.0.4            </span></span>
<span><span class="co">#&gt;  [11] vctrs_0.6.1                   stringr_1.5.0                </span></span>
<span><span class="co">#&gt;  [13] pkgconfig_2.0.3               crayon_1.5.2                 </span></span>
<span><span class="co">#&gt;  [15] fastmap_1.1.1                 ellipsis_0.3.2               </span></span>
<span><span class="co">#&gt;  [17] XVector_0.39.0                rmdformats_1.0.4             </span></span>
<span><span class="co">#&gt;  [19] utf8_1.2.3                    promises_1.2.0.1             </span></span>
<span><span class="co">#&gt;  [21] rmarkdown_2.21                grDevices_4.3.0              </span></span>
<span><span class="co">#&gt;  [23] sessioninfo_1.2.2             tzdb_0.3.0                   </span></span>
<span><span class="co">#&gt;  [25] strawr_0.0.91                 ragg_1.2.5                   </span></span>
<span><span class="co">#&gt;  [27] purrr_1.0.1                   bit_4.0.5                    </span></span>
<span><span class="co">#&gt;  [29] xfun_0.38                     zlibbioc_1.45.0              </span></span>
<span><span class="co">#&gt;  [31] cachem_1.0.7                  GenomeInfoDb_1.35.16         </span></span>
<span><span class="co">#&gt;  [33] jsonlite_1.8.4                blob_1.2.4                   </span></span>
<span><span class="co">#&gt;  [35] later_1.3.0                   rhdf5filters_1.11.2          </span></span>
<span><span class="co">#&gt;  [37] DelayedArray_0.25.0           interactiveDisplayBase_1.37.0</span></span>
<span><span class="co">#&gt;  [39] Rhdf5lib_1.21.0               BiocParallel_1.33.12         </span></span>
<span><span class="co">#&gt;  [41] parallel_4.3.0                R6_2.5.1                     </span></span>
<span><span class="co">#&gt;  [43] bslib_0.4.2                   stringi_1.7.12               </span></span>
<span><span class="co">#&gt;  [45] reticulate_1.28               GenomicRanges_1.51.4         </span></span>
<span><span class="co">#&gt;  [47] jquerylib_0.1.4               Rcpp_1.0.10                  </span></span>
<span><span class="co">#&gt;  [49] bookdown_0.33                 SummarizedExperiment_1.29.1  </span></span>
<span><span class="co">#&gt;  [51] knitr_1.42                    IRanges_2.33.0               </span></span>
<span><span class="co">#&gt;  [53] httpuv_1.6.9                  Matrix_1.5-3                 </span></span>
<span><span class="co">#&gt;  [55] tidyselect_1.2.0              yaml_2.3.7                   </span></span>
<span><span class="co">#&gt;  [57] codetools_0.2-19              curl_5.0.0                   </span></span>
<span><span class="co">#&gt;  [59] lattice_0.20-45               tibble_3.2.1                 </span></span>
<span><span class="co">#&gt;  [61] withr_2.5.0                   KEGGREST_1.39.0              </span></span>
<span><span class="co">#&gt;  [63] shiny_1.7.4                   InteractionSet_1.27.0        </span></span>
<span><span class="co">#&gt;  [65] Biobase_2.59.0                basilisk.utils_1.11.2        </span></span>
<span><span class="co">#&gt;  [67] evaluate_0.20                 desc_1.4.2                   </span></span>
<span><span class="co">#&gt;  [69] Biostrings_2.67.2             pillar_1.9.0                 </span></span>
<span><span class="co">#&gt;  [71] BiocManager_1.30.20           filelock_1.0.2               </span></span>
<span><span class="co">#&gt;  [73] MatrixGenerics_1.11.1         stats4_4.3.0                 </span></span>
<span><span class="co">#&gt;  [75] plotly_4.10.1                 generics_0.1.3               </span></span>
<span><span class="co">#&gt;  [77] vroom_1.6.1                   rprojroot_2.0.3              </span></span>
<span><span class="co">#&gt;  [79] RCurl_1.98-1.12               BiocVersion_3.17.1           </span></span>
<span><span class="co">#&gt;  [81] S4Vectors_0.37.4              ggplot2_3.4.2                </span></span>
<span><span class="co">#&gt;  [83] munsell_0.5.0                 scales_1.2.1                 </span></span>
<span><span class="co">#&gt;  [85] xtable_1.8-4                  glue_1.6.2                   </span></span>
<span><span class="co">#&gt;  [87] lazyeval_0.2.2                tools_4.3.0                  </span></span>
<span><span class="co">#&gt;  [89] datasets_4.3.0                BiocIO_1.9.2                 </span></span>
<span><span class="co">#&gt;  [91] data.table_1.14.8             fs_1.6.1                     </span></span>
<span><span class="co">#&gt;  [93] rhdf5_2.43.2                  grid_4.3.0                   </span></span>
<span><span class="co">#&gt;  [95] tidyr_1.3.0                   crosstalk_1.2.0              </span></span>
<span><span class="co">#&gt;  [97] AnnotationDbi_1.61.2          colorspace_2.1-0             </span></span>
<span><span class="co">#&gt;  [99] GenomeInfoDbData_1.2.10       basilisk_1.11.3              </span></span>
<span><span class="co">#&gt; [101] cli_3.6.1                     rappdirs_0.3.3               </span></span>
<span><span class="co">#&gt; [103] textshaping_0.3.6             fansi_1.0.4                  </span></span>
<span><span class="co">#&gt; [105] viridisLite_0.4.1             dplyr_1.1.1                  </span></span>
<span><span class="co">#&gt; [107] gtable_0.3.3                  sass_0.4.5                   </span></span>
<span><span class="co">#&gt; [109] digest_0.6.31                 htmlwidgets_1.6.2            </span></span>
<span><span class="co">#&gt; [111] memoise_2.0.1                 htmltools_0.5.5              </span></span>
<span><span class="co">#&gt; [113] pkgdown_2.0.7                 lifecycle_1.0.3              </span></span>
<span><span class="co">#&gt; [115] here_1.0.1                    httr_1.4.5                   </span></span>
<span><span class="co">#&gt; [117] mime_0.12                     bit64_4.0.5</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Jacques Serizay.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
